bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/findandreplace/IFindReplaceLogic.java

```diff
@@ -0,0 +1,183 @@
+	 *                      current thread will be used. If there is no Display for
+	 *                      the current thread,the runnable code will be executed
+	 *                      and no busy feedback will be displayed.y
+	 */
+	void performReplaceAll(String findString, String replaceString, Display display);
+
+	/**
+	 * Selects all occurrences of findString.
+	 *
+	 * @param findString The String to find and select
+	 * @param display    The UI's Display The UI's Display
+	 */
+	void performSelectAll(String findString, Display display);
+
+	/**
+	 * Locates the user's findString in the target
+	 *
+	 * @param searchString the String to search for
+	 * @return Whether the string was found in the target
+	 *
+	 * @since 3.7
+	 */
+	boolean performSearch(String searchString);
+
+	/**
+	 * @param findString    the String to select as part of the search
+	 * @param caseSensitive Should case be respected in searching?
+	 * @param wholeWord     Should the find String be searched as a whole word?
+	 * @param regExSearch   Should the findString be interpreted as RegEx?
+	 * @return The amount of selected Elements
+	 */
+	int selectAll(String findString, boolean caseSensitive, boolean wholeWord, boolean regExSearch);
+
+	/**
+	 * Returns the position of the specified search string, or <code>-1</code> if
+	 * the string can not be found when searching using the given options.
+	 *
+	 * @param findString    the string to search for
+	 * @param startPosition the position at which to start the search
+	 * @return the occurrence of the find string following the options or
+	 *         <code>-1</code> if nothing found
+	 * @since 3.0
+	 */
+	int findIndex(String findString, int startPosition);
+
+	/**
+	 * Searches for a string starting at the given offset and using the specified
+	 * search directives. If a string has been found it is selected and its start
+	 * offset is returned.
+	 *
+	 * @param offset        the offset at which searching starts
+	 * @param findString    the string which should be found
+	 * @return the position of the specified string, or -1 if the string has not
+	 *         been found
+	 * @since 3.0
+	 */
+	int findAndSelect(int offset, String findString);
+
+	/**
+	 * Replaces the selection and jumps to the next occurrence of findString
+	 * instantly. Will not fail in case the selection is invalidated, eg. after a
+	 * replace operation or after the target was updated
+	 *
+	 * @param findString    the string to replace
+	 * @param replaceString the string to put in place of findString
+	 * @return whether a replacement has been performed
+	 */
+	boolean performReplaceAndFind(String findString, String replaceString);
+
+	boolean performSelectAndReplace(String findString, String replaceString);
+
+	void updateTarget(IFindReplaceTarget newTarget, boolean canEditTarget);
+
+	void dispose();
+
+	/**
+	 * Returns whether the target is editable.
+	 *
+	 * @return <code>true</code> if target is editable
+	 */
+	boolean isEditable();
```

Comment:
This method does not belong to `IFindReplaceLogic` as it does not return if the logic is editable. The method must either be called `isTargetEditable()`, or, better, removed, as the `target` is exposed anyway, to every user can check whether the target is editable, so no need to make this interface more verbose.